### PR TITLE
avm2: Support any name in XML get_property_local

### LINKED ISSUE #9999: Into space 2 - white screen when clicking play
### Describe the bug

```
ruffle_core::loader: Error during movie loading: FetchError("failed to connect to the server")
ERROR ruffle_core::loader: Error during movie loading: FetchError("failed to connect to the server")
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: TypeError, ptr: 0x55934a9cb9b0 }))
ERROR ruffle_core::avm2::activation: TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: addEventListener)
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "addedToStage", class: flash.events::Event, ptr: 0x55934a9cb580 }) to handler FunctionObject(FunctionObject { ptr: 0x55934a9844e0 }) : AvmError(Object(ErrorObject(ErrorObject { class: TypeError, ptr: 0x55934a9cb9b0 })))
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: TypeError, ptr: 0x55934aa11be0 }))
ERROR ruffle_core::avm2::activation: TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: addEventListener)
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "addedToStage", class: flash.events::Event, ptr: 0x55934aa11760 }) to handler FunctionObject(FunctionObject { ptr: 0x55934a9d2330 }) : AvmError(Object(ErrorObject(ErrorObject { class: TypeError, ptr: 0x55934aa11be0 })))
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::activation: AVM2 error: Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 }))
ERROR ruffle_core::avm2::activation: ReferenceError: Error #1069: Property * not found on XML and there is no default value.
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "complete", class: flash.events::Event, ptr: 0x55934a43c280 }) to handler FunctionObject(FunctionObject { ptr: 0x55934a485330 }) : AvmError(Object(ErrorObject(ErrorObject { class: ReferenceError, ptr: 0x55934acc1d80 })))

```

### Expected behavior

the game should load

### Affected platform

Desktop app

### Operating system

Windows 10

### Browser

_No response_

### Additional information

[603934_IntoSpace2.swf.zip](https://github.com/ruffle-rs/ruffle/files/10949383/603934_IntoSpace2.swf.zip)

### PR DESCRIPTION
**Pull request by evilpie**
Fixes #9999

### CODE REVIEW

**Comment by Aaron1011** [comment]
Thanks for the PR! Could you add a test for this?

File: core/src/avm2/object/xml_object.rs [hunk 1]
```diff
@@ -118,35 +118,34 @@ impl<'gc> TObject<'gc> for XmlObject<'gc> {
                         return Ok(Value::Undefined);
                     }
                 }
+            }
 
-                let matched_children = if let E4XNodeKind::Element {
```
Comment (Aaron1011) [inline core/src/avm2/object/xml_object.rs:122]: Is this just a formatting change? If so, could you revert this file back to the original state?

Comment (evilpie) [reply core/src/avm2/object/xml_object.rs:122]: No the indentation changed. local_name is None for any.

**Comment by Lord-McSweeney** [comment]
If this PR adds support for `XML.child("*")`, could you add a test for it?

File: core/src/avm2/object/xml_object.rs (hunk 1 above)
Comment (Aaron1011) [reply core/src/avm2/object/xml_object.rs:122]: I see - thanks

**Comment by Aaron1011** [comment]
@Lord-McSweeney That can be done in a follow-up

**Comment by evilpie** [comment]
Thanks for the reminder, wildcard support is in #10041.

**Pull request merged**